### PR TITLE
Feat: Added historyPage

### PR DESCRIPTION
**Pull request by RahulSingh9131**
This branch contains
1. history page 
2. context and reducer

### CODE REVIEW

File: src/css/screens/videocard.css [hunk 1]
```diff
@@ -61,4 +61,18 @@
...
 .filled-like{
     color: var(--red-color);
+}
+.videocard-play{
+    position: absolute;
+    right: 140px;
+    top: 50px;
+    cursor: pointer;
+    width: 40px !important;
+    height: 40px !important;
```
Comment (Sameony) [inline src/css/screens/videocard.css:71]: important in css is discouraged, do try to avoid using it in direct styles. Why is it used here by the way?

**Review by shuntzhere — APPROVED** [review]
The code looks good. The classes are really understandable too.

**Review by Sameony — APPROVED** [review]
Good pr, the code is very clean and concise
liked the structured reducers

File: src/components/VideoCard.js [hunk 2]
```diff
@@ -30,6 +34,13 @@ function VideoCard(videos) {
   return (
     <div className='videocard' key={_id}>
         <img className='videocard-thumbnail' src={image} alt={title}/>
+        {
+          historyBasket.some((c)=>c._id===videos._id)?(
+            <PlayArrowIcon className='videocard-play filled-play'/>
+          ):(
+            <PlayCircleIcon className='videocard-play' onClick={()=>historyDispatch({type:"ADD_TO_HISTORY",payload:videos})}/>
```
Comment (Vishruta-Patil) [inline src/components/VideoCard.js:41]: For action type  use constants

File: src/reducer/historyReducerFunc.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export const historyReducerFunc=(historyState,action)=>{
+    switch(action.type){
+        case "ADD_TO_HISTORY":
+            return {...historyState,
+                historyBasket:[...historyState.historyBasket,action.payload]};
+        case "REMOVE_FROM_HISTORY":
+            return {...historyState,
+                historyBasket:historyState.historyBasket.filter((c)=>c._id!==action.payload._id)}
```
Comment (Vishruta-Patil) [inline src/reducer/historyReducerFunc.js:8]: same as above, make a separate file for action type

**Review by Vishruta-Patil — APPROVED** [review]
Other than that everything looks good to merge

File: src/css/screens/videocard.css (hunk 1 above)
Comment (RahulSingh9131) [reply src/css/screens/videocard.css:71]: to overwrite styles of material icons, one has to use !important.

**Pull request merged**